src/Agent.Worker/TaskRunner.cs

```diff
@@ -564,5 +568,31 @@ private void PrintTaskMetaData(Definition taskDefinition)
             ExecutionContext.Output($"Help         : {taskDefinition.Data.HelpUrl ?? taskDefinition.Data.HelpMarkDown}");
             ExecutionContext.Output("==============================================================================");
         }
+        private void PublishTelemetry(Definition taskDefinition, HandlerData handlerData)
+        {
+            ArgUtil.NotNull(Task, nameof(Task));
+            ArgUtil.NotNull(Task.Reference, nameof(Task.Reference));
+            ArgUtil.NotNull(taskDefinition.Data, nameof(taskDefinition.Data));
+
+            var useNode10 = AgentKnobs.UseNode10.GetValue(ExecutionContext).AsString();
+            Dictionary<string, string> telemetryData = new Dictionary<string, string>();
+            telemetryData.Add("Task Name", Task.Reference.Name);
+            telemetryData.Add("Task Id", Task.Reference.Id.ToString());
+            telemetryData.Add("Version", Task.Reference.Version);
+            telemetryData.Add("OS", PlatformUtil.HostOS.ToString());
+            telemetryData.Add("Expected Execution Handler", string.Join(", ", taskDefinition.Data.Execution.All));
+            telemetryData.Add("Real Execution Handler", handlerData.ToString());
+            telemetryData.Add("UseNode10", useNode10);
+
+            var publishTelemetryCmd = new TelemetryCommandExtension();
+            publishTelemetryCmd.Initialize(HostContext);
+
+            var json = JsonConvert.SerializeObject(telemetryData, Formatting.None);
+            var cmd = new Command("telemetry", "publish");
+            cmd.Data = json;
+            cmd.Properties.Add("area", "PipelinesTasks");
+            cmd.Properties.Add("feature", "ExecutionHandler");
+            publishTelemetryCmd.ProcessCommand(ExecutionContext, cmd);
```

Comment:
Maybe it would be better to add extra blank line before this line and move here lines with creating and initializing of publishTelemetryCmd

Reply:
Reasonable, done